CorsixTH/Lua/hospitals/player_hospital.lua

```diff
@@ -332,6 +334,27 @@ function PlayerHospital:msgKilled()
   end
 end
 
+--! Once a month the advisor may warn about long queues.
+--! Rooms requiring a doctor occasionally trigger the generic message
+function PlayerHospital:warnForLongQueues()
+  local busy_rooms = {}
+  for _, room in pairs(self.world.rooms) do
+    if #room.door.queue > 5 then
+      busy_rooms[#busy_rooms + 1] = room.room_info
+    end
+  end
+  if #busy_rooms == 0 then return end
+  local chosen_room = busy_rooms[math.random(1, #busy_rooms)]
+  -- Required staff that is not nurse is doctor, researcher, surgeon or psych
+  if chosen_room.required_staff and not chosen_room.required_staff["Nurse"]
+      and math.random(1, 3) > 1 then
+    self.world.ui.adviser:say(
+        _A.warnings.queue_too_long_send_doctor:format(chosen_room.name))
+  else
+    self.world.ui.adviser:say(_A.warnings.queues_too_long)
+  end
```

Comment:
Not sure we should guarantee an adviser pop-up every month from long queues if the player is constantly having long queues.